discovery-provider/integration_tests/tasks/test_anchor_program_indexer.py

```diff
@@ -111,6 +122,62 @@ def test_parse_tx(app):
     assert owner_eth_address_hex == "0x25A3Acd4758Ab107ea0Bd739382B8130cD1F204d"
 
 
+@pytest.mark.asyncio
+async def test_fetch_metadata(app, mocker):
+    with app.app_context():
+        db = get_db()
+        redis = get_redis()
+
+    populate_mock_db(db, basic_entities, block_offset=3)
```

Comment:
are you able to get db since the rest of this execution isn't in the same block scope of the `with` stuff?

Reply:
yea that's just how python scoping